src/nnet3/nnet-am-decodable-simple.h

```diff
@@ -213,6 +219,10 @@ class NnetDecodableBase {
   // online_ivector_period_ helps us interpret online_ivector_feats_; it's the
   // number of frames the rows of ivector_feats are separated by.
   int32 online_ivector_period_;
+  // In the multiple ivectors case, it is the interval between two consecutive
+  // frames that we want to get ivectors for, so that we can dump multiple
+  // ivectors for a chunk. In the single ivector case, its value is just 0.
```

Comment:
I think you are assuming that in the case of single ivector case, i.e., when ReplaceIndex is used the value is always going to be 0. Is this correct ? If yes this is the wrong assumption.

Even in the case you force GetTimerPeriodForIvector() to return 0 in the single ivector case, I think it is bad design. Please determine single/multiple ivector case just based on whether ivector_/online_ivectors_ is NULL.